client/auth.go

```diff
@@ -0,0 +1,32 @@
+package client
+
+// Handle the "USER" command.
+func (c *Handler) handleUSER() {
+	c.user = c.param
+	c.WriteMessage(StatusUserOK, "User name okay, need password.")
+}
+
+// Handle the "PASS" command.
+func (c *Handler) handlePASS() {
+	if c.user == "" {
+		c.WriteMessage(StatusBadCommandSequence, "User is expected before Pass")
+		return
+	}
+
+	defer func() {
+		c.user = ""
+	}()
+
+	username := c.user
+	password := c.param
+
+	if _, ok := c.serverSetting.Users[username]; ok {
+		if username == "anonymous" || password == c.serverSetting.Users[username] {
```

Comment:
We can use the value of map directly.
```suggestion
	if v, ok := c.serverSetting.Users[username]; ok {
		if username == "anonymous" || password == v {
```